### DISCUSSION TITLE
Why is my list scrolling super fast after scene change.

### DISCUSSION THREAD

**Discussion by lavizp** [LanKuDot/game_modules#30] [Q&A] [answered]
I have setup a circular scroll in my main menu, The canvas for the main menu is saved as a DontDestroyOnLoad to save menu properties. When I change my scene and change back to the menu while playing the game, I reassign the main camera through script:
 `canvas.GetComponentInParent<Canvas>().worldCamera = Camera.main;
`
After that, the list scrolls super fast, no errors are thrown. Please let me know if I can fix this bug.

**Comment by LanKuDot** [comment, marked as answer]
Hi, the list stores the referencing camera of the canvas in the initialization.
And the list uses it to calculate the local position by calling `RectTransformUtility.ScreenPointToLocalPointInRectangle()`, which the null reference of the camera parameter is acceptable. That's why there has no error.
But it will treat the canvas that it is in the screen space - overlay mode.
Therefore, you also have to reassign the camera into the list.

Try this:
1. At the [line 47](https://github.com/LanKuDot/game_modules/blob/5a39d36234d08cd47536de52585f8b691ce2a457/CircularScrollingList/Scripts/ListPositionCtrl.cs#L47) in the `ListPositionCtrl.cs`, remove `readonly` keyword from the `_canvasRefCamera`. 
2. Add a new function to the `ListPositionCtrl.cs`:

```csharp
public void SetCanvasRefCamera(Camera camera)
{
    _canvasRefCamera = camera;
}
```

3. Add a new function to the `CircularScrollingList.cs`:

```csharp
public void ReassignCanvasRefCamera()
{
    GetComponentReference()
    _listPositionCtrl.SetCanvasRefCamera(_canvasRefCamera);
}
```

4. Then, call `CircularScrollingList.ReassignCanvasRefCamera()` after you assign the camera to the canvas.

**Reply by lavizp** [reply]
Thank you so much, you are a genius,
